crates/net/discv5/src/lib.rs

```diff
@@ -105,20 +113,27 @@ pub trait HandleDiscv5 {
 
         Ok(NodeRecord { address: udp_socket.ip(), tcp_port, udp_port: udp_socket.port(), id })
     }
+
+    /// Applies filtering rules on an ENR. Returns [`Ok`](FilterOutcome::Ok) if peer should be
+    /// passed up to app, and [`Ignore`](FilterOutcome::Ignore) if peer should instead be dropped.
+    fn filter_discovered_peer(&self, enr: &discv5::Enr) -> FilterOutcome;
 }
 
 /// Transparent wrapper around [`discv5::Discv5`].
 #[derive(Deref, DerefMut, Clone, Constructor)]
-pub struct DiscV5 {
+pub struct DiscV5<T = DefaultFilter> {
     #[deref]
     #[deref_mut]
```

Comment:
do we want to support swapping this out?

Reply:
yes definitely. it basically exposes the api of sigp/discv5 which supports injecting predicates to constrain FINDNODE queries.